latch/services/execute.py

```diff
@@ -0,0 +1,204 @@
+from flytekit.core.type_engine import TypeEngine
+from latch.utils import retrieve_or_login
+
+
+def execute(params_file: Path, version: Union[None, str] = None) -> bool:
+    """Executes a versioned workflow with parameters specified in python.
+
+    Args:
+        params_file: A path pointing to a python file with a function call
+            representing the workflow execution with valid parameter values.
+        version: An optional workflow version to execute, defaults to latest if
+            None is provided
+
+        Parameters for the workflow execution are specified in plain python as
+        a function call, as the syntax is highly legible and easier to work
+        with than alternatives such as YAML, JSON. This file is passed to the
+        `execute` service and should contain no code other than the single
+        call.
+
+    Returns:
+        True if successful else False
+
+    Example: ::
+
+        # A minimal parameter file used to execute the boilerplate
+
+        from latch.types import LatchFile
+
+        params = {
+            "_name": "wf.assemble_and_sort",
+            "read1": LatchFile("latch:///read1"),
+            "read2": LatchFile("latch:///read2"),
+        }
+    """
+
+    token = retrieve_or_login()
+
+    with open(params_file, "r") as pf:
+
+        param_code = pf.read()
+        param_module = ModuleType("wf_params")
+        exec(param_code, param_module.__dict__)
+
+        module_vars = vars(param_module)
+        try:
+            wf_params = module_vars["params"]
+        except KeyError as e:
+            raise ValueError(
+                f"Execution file {params_file.name} needs to have"
+                " a parameter value dictionary named 'params'"
+            ) from e
+
+        wf_name = wf_params.get("_name")
+        if wf_name is None:
+            raise ValueError(
+                f"The dictionary of parameters in the launch file lacks the"
+                " _name key used to identify the workflow. Make sure a _name"
+                " key with the workflow name exists in the dictionary."
+            )
+
+        wf_id, wf_interface = _get_workflow_interface(token, wf_name, version)
+
+        wf_vars = wf_interface["variables"]
+        wf_literals = {}
+        for key, value in wf_vars.items():
+
+            ctx = FlyteContextManager.current_context()
+            literal_type_json = value["type"]
+            literal_type = gpjson.Parse(json.dumps(literal_type_json), LiteralType())
+
+            if key in wf_params:
+
+                python_value = wf_params[key]
+                # Recover parameterized generics for TypeTransformer.
+                python_type = _to_python_literal(python_value)
+
+                python_type_literal = TypeEngine.to_literal(
+                    ctx, python_value, python_type, literal_type
+                )
+
+                wf_literals[key] = json.loads(
+                    gpjson.MessageToJson(python_type_literal.to_flyte_idl())
```

Comment:
```suggestion
                    gpjson.MessageToDict(python_type_literal.to_flyte_idl())
```

and del the `json.loads`